save-backend/src/main/kotlin/com/saveourtool/save/backend/controllers/vulnerability/VulnerabilityController.kt

```diff
@@ -57,23 +57,17 @@ class VulnerabilityController(
         @RequestBody filters: VulnerabilityFilter,
         authentication: Authentication?,
     ): Mono<VulnerabilityDtoList> = blockingToMono {
-        if (authentication?.hasRole(Role.SUPER_ADMIN) != true && filters.status != VulnerabilityStatus.APPROVED && !filters.isOwner) {
+        if (
+            // if user is not authenticated, he will have authentication = null and will not get other's submitted vulnerabilities
+            filters.status != VulnerabilityStatus.APPROVED && authentication?.name != filters.authorName &&
+                    // only if user is NOT admin, if admin - everything is fine
+                    authentication?.hasRole(Role.SUPER_ADMIN) == false
```

Comment:
Maybe a bit more readable way is `authentication?.hasRole(Role.SUPER_ADMIN) != true` as thus we can easily see that when `authentication?.hasRole(Role.SUPER_ADMIN) == null` everything is fine (just a point with readability)

Reply:
I thought in this way, but here is the logic is that you don't have SUPER ADMIN role (your role superadmin is false, like a flag)